src/sparseml/pytorch/torchvision/train.py

```diff
@@ -415,21 +417,51 @@ def collate_fn(batch):
         alpha = 1.0 - args.model_ema_decay
         alpha = min(1.0, alpha * adjust)
         model_ema = utils.ExponentialMovingAverage(
-            model_without_ddp, device=device, decay=1.0 - alpha
+            model, device=device, decay=1.0 - alpha
         )
 
-    if args.resume:
-        checkpoint = torch.load(args.resume, map_location="cpu")
-        model_without_ddp.load_state_dict(checkpoint["model"])
-        if not args.test_only:
-            optimizer.load_state_dict(checkpoint["optimizer"])
-            lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
+    if args.checkpoint_path:
+        checkpoint = _load_checkpoint(args.checkpoint_path)
+        model.load_state_dict(checkpoint["model"])
+        optimizer.load_state_dict(checkpoint["optimizer"])
+        lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
+        if model_ema:
+            model_ema.load_state_dict(checkpoint["model_ema"])
+        if scaler:
+            scaler.load_state_dict(checkpoint["scaler"])
+
         args.start_epoch = checkpoint["epoch"] + 1
+
+        manager = ScheduledModifierManager.from_yaml(args.recipe_path)
+        checkpoint_manager = ScheduledModifierManager.from_yaml(
+            checkpoint["checkpoint_recipe"]
+        )
+
+        checkpoint_manager.apply_structure(model, epoch=checkpoint["epoch"])
+    elif args.resume:
+        checkpoint = _load_checkpoint(args.resume)
+        model.load_state_dict(checkpoint["model"])
+        optimizer.load_state_dict(checkpoint["optimizer"])
+        lr_scheduler.load_state_dict(checkpoint["lr_scheduler"])
         if model_ema:
             model_ema.load_state_dict(checkpoint["model_ema"])
         if scaler:
             scaler.load_state_dict(checkpoint["scaler"])
 
+        args.start_epoch = checkpoint["epoch"] + 1
+
+        # NOTE: override manager with the checkpoint's manager
+        manager = ScheduledModifierManager.from_yaml(checkpoint["checkpoint_recipe"])
+        checkpoint_manager = None
+
+        manager.initialize(model, epoch=checkpoint["epoch"])
+    else:
+        manager = ScheduledModifierManager.from_yaml(args.recipe_path)
+        checkpoint_manager = None
+
+    # TODO do we still need this in case of resume or checkpoint_path?
+    optimizer = manager.modify(model, optimizer, len(data_loader))
```

Comment:
@bfineran still have a question here on if this should be done for all three cases (resume, checkpoint_path, start from scratch)

Reply:
in resume, we should be using the initialized checkpoint manager (can call modify up there with the optimizer)

Reply:
Yep in resume the manager gets set to the checkpoint's recipe (i.e. the checkpoint manager)